File: typescript/problems/p0014_longest_common_prefix/solution.test.ts
import longestCommonPrefix from "./solution";

test("test solution", () => {
  expect(longestCommonPrefix(["flower", "flow", "flight"])).toBe("fl");
  expect(longestCommonPrefix(["dog", "racecar", "car"])).toBe("");
  expect(longestCommonPrefix(["ab", "a"])).toBe("a");
});

File: java/problems/p0014_longest_common_prefix/SolutionTest.java
package problems.p0014_longest_common_prefix;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class SolutionTest {
    @Test
    void testSolution() {
        Solution solution = new Solution();
        assertEquals("fl", solution.longestCommonPrefix(new String[]{"flower", "flow", "flight"}));
        assertEquals("", solution.longestCommonPrefix(new String[] { "dog", "racecar", "car" }));
        assertEquals("a", solution.longestCommonPrefix(new String[] { "ab", "a" }));
    }
}
